content/docs/self-hosted-runners.md

```diff
@@ -448,7 +448,101 @@ The same credentials can also be used for
 
 </admon>
 
-### On-premise (Local) Runners
+## GitLab CI/CD and container images from private registries
```

Comment:
this is GL specific? what about GH?

Reply:
This is GitLab-specific. On GitHub it's not possible.[^1]

The closest approximation I can think of is this (not working) example:

```yaml
on: push
jobs:
  authentication:
    runs-on: ubuntu-latest
    outputs:
      username: ${{ steps.authenticate.outputs.username }}
      password: ${{ steps.authenticate.outputs.password }}
    steps:
      - id: authenticate
        run: |
          echo "username=AWS" >> $GITHUB_OUTPUT
          echo "password=$(aws ecr get-login-password --region us-east-1)" >> $GITHUB_OUTPUT
  example:
    needs: authentication
    runs-on: ubuntu-latest
    container:
      image: node
      credentials:
        username: ${{ needs.authentication.outputs.username }}
        password: ${{ needs.authentication.outputs.password }}
    steps:
      - run: true
```

[^1]: Read as «not easy enough to be documented»

Reply:
The closest approximation I can think of is this (not working) example:

```yaml
on: push
jobs:
  authentication:
    runs-on: ubuntu-latest
    outputs:
      username: ${{ steps.authenticate.outputs.username }}
      password: ${{ steps.authenticate.outputs.password }}
    steps:
      - id: authenticate
        run: |
          echo "username=AWS" >> $GITHUB_OUTPUT
          echo "password=$(aws ecr get-login-password --region us-east-1)" >> $GITHUB_OUTPUT
  example:
    needs: authentication
    runs-on: ubuntu-latest
    container:
      image: node
      credentials:
        username: ${{ needs.authentication.outputs.username }}
        password: ${{ needs.authentication.outputs.password }}
    steps:
      - run: true
```

Reply:
...or https://agileek.github.io/software/aws/2020/10/28/using-an-ecr-image-in-github-actions

Reply:
kewl. I think this deserves a new page as it's not really about self-hosted runners, right?

Reply:
It _is_ about self-hosted runners, on GitLab, with private container registries. Definition of niceh.